lib/elixir/src/elixir_errors.erl

```diff
@@ -57,6 +61,131 @@ emit_diagnostic(Severity, Position, File, Message, Stacktrace) ->
 
   ok.
 
+% --------- Fancy Diagnostics
+
+fancy_exception_snippet(File, LineNumber, Column, Description, Snippet) ->
+  #{content := Content, offset := Offset} = Snippet,
+
+  LineDigits = get_line_number_digits(LineNumber),
+  Spacing = n_spaces(LineDigits + 1),
+
+  Formatted = io_lib:format(
+    " ~ts┌─ ~ts~ts\n"
+    " ~ts│\n"
+    " ~p │ ~ts\n"
+    " ~ts│ ~ts\n"
+    " ~ts│\n"
+    " ~ts~ts",
+    [
+     Spacing, prefix(error), file_format({LineNumber, Column}, File),
+     Spacing,
+     LineNumber, Content,
+     Spacing, highlight_below_line(Content, Offset, error),
+     Spacing,
+     Spacing, format_message(Description, LineDigits)
```

Comment:
No padding for the message. Just let it roll over and cover multiple lines if necessary. Especially because we don't guarantee our error message wrap at a given limit.

```suggestion
     Spacing, Description
```

Reply:
Currently format_message just pads the new lines of the message, so if there's no newline the message with stay long and will be auto-wrapped by editor/terminal, which is great, and if the message was designed with line breaks in mind, it will stay alilgned with the error.

That's useful mostly for warnings, where there some type/guard warnings that span multiple lines, with this they look like:
```
   |
   |
   warnings: guard foo bar
   
   expected type:
     atom()
   but rather got:
     string()     
```

Without using it they'll look misaligned:
```
   |
   |
   warnings: guard foo bar

expected type:
  atom()
but rather got:
  string()     
```

I don't remember of seeing any syntax/missingtoken messages that span multiple lines, so I'll double check and remove it if safe, but I think that the padding will very welcome to keep styling when implementing the fancy diagnostics.